modules/installation-azure-user-defined-routing.adoc

```diff
@@ -0,0 +1,84 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_azure/installing-azure-private.adoc
+
+[id="installation-azure-user-defined-routing_{context}"]
+= User-defined outbound routing
+
+In {product-title}, you can choose your own outbound routing for a cluster to
+connect to the internet. This allows you to skip the creation of public IP
+addresses and the public load balancer.
+
+You can configure user-defined routing by modifying parameters in the
+`install-config.yaml` file before installing your cluster. A pre-existing VNet
+is required to use outbound routing when installing a cluster; the installation
+program is not responsible for configuring this.
+
+When configuring a cluster to use user-defined routing, the installation program
+does not create the following resources:
+
+* Outbound rules for access to Internet.
+* Public IPs for the public load balancer.
+* Kubernetes Service object to add the cluster machines to the public load
+balancer for outbound requests.
+
+You must ensure the following items are available before setting user-defined
+routing:
+
+* Egress to the internet is possible to pull container images, unless using an
+internal registry mirror.
+* The cluster can access Azure APIs.
+* Various allowlist endpoints are configured. You can reference these endpoints
+in the *Configuring your firewall* section.
+
+There are several pre-existing networking setups that are supported for internet
+access using user-defined routing.
+
+[discrete]
+== Private cluster with network address translation
+
+You can use link:https://docs.microsoft.com/en-us/azure/virtual-network/nat-overview[Azure VNET network address translation (NAT)]
+to provide outbound internet access for the subnets in your cluster. You can
+reference
+link:https://docs.microsoft.com/en-us/azure/virtual-network/quickstart-create-nat-gateway-cli[Create a NAT gateway using Azure CLI]
+in the Azure documentation for configuration instructions.
+
+When using a VNet setup with Azure NAT and user-defined routing configured, you
+can create a private cluster with no public endpoints. 
+
+[discrete]
+== Private cluster with Azure Firewall
+
+You can use Azure Firewall to provide outbound routing for the VNet used to
+install the cluster. You can learn more about
+link:https://docs.microsoft.com/en-us/azure/aks/egress-outboundtype#deploy-a-cluster-with-outbound-type-of-udr-and-azure-firewall[providing user-defined routing with Azure Firewall]
+in the Azure documentation.
+
+When using a VNet setup with Azure Firewall and user-defined routing configured,
+you can create a private cluster with no public endpoints.
+
+[discrete]
+== Private cluster with a proxy configuration
+
+You can use a proxy with user-defined routing to allow egress to the internet.
+You must make sure that cluster Operators do not access Azure APIs using a
```

Comment:
For some reason, "You must make sure" read funny to me. Maybe "You must ensure" ?